File: client-app/src/app/shared/models/network/Link.ts
import {Status} from './Status';

export class Link {
  private readonly _id: number;
  private _weight: number;
  private _status: Status;
  private readonly _sourceId: number;
  private readonly _destinationId: number;


  constructor(id: number, weight: number, status: Status, sourceId: number, destinationId: number) {
    this._id = id;
    this._weight = weight;
    this._status = status;
    this._sourceId = sourceId;
    this._destinationId = destinationId;
  }

  get weight(): number {
    return this._weight;
  }

  set weight(value: number) {
    this._weight = value;
  }

  get status(): Status {
    return this._status;
  }

  set status(value: Status) {
    this._status = value;
  }

  get id(): number {
    return this._id;
  }

  get sourceId(): number {
    return this._sourceId;
  }

  get destinationId(): number {
    return this._destinationId;
  }
}

File: network-configuration-server/src/main/java/en/ubb/networkconfiguration/persistence/domain/network/runtime/Link.java
package en.ubb.networkconfiguration.persistence.domain.network.runtime;

import en.ubb.networkconfiguration.persistence.domain.BaseEntity;
import en.ubb.networkconfiguration.persistence.domain.network.virtual.VirtualLink;
import lombok.*;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "outputLinks")
public class Link extends BaseEntity<Long> {

    @Column(name = "weight", nullable = false)
    private double weight;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "source_id")
    private Node source;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "destination_id")
    private Node destination;

    @OneToMany(mappedBy = "link", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<VirtualLink> virtualLinks = new ArrayList<>();

    @Builder(toBuilder = true)
    public Link(Long id, double weight, Node source, Node destination, List<VirtualLink> virtualLinks) {
        super(id);
        this.weight = weight;
        this.source = source;
        this.destination = destination;
        this.virtualLinks = virtualLinks;
    }

    public Link(Link link) {
        this.weight = link.getWeight();
    }
}
